net/peer.go

```diff
@@ -97,75 +81,20 @@ func NewPeer(
 		opt(options)
 	}
 
-	connManager, err := connmgr.NewConnManager(100, 400, connmgr.WithGracePeriod(time.Second*20))
-	if err != nil {
-		return nil, err
-	}
-
-	var listenAddresses []multiaddr.Multiaddr
-	for _, addr := range options.ListenAddresses {
-		listenAddress, err := multiaddr.NewMultiaddr(addr)
-		if err != nil {
-			return nil, err
-		}
-		listenAddresses = append(listenAddresses, listenAddress)
-	}
-
-	ctx, cancel := context.WithCancel(ctx)
-	defer func() {
-		if p == nil {
-			cancel()
-		}
-	}()
-
-	peerstore, err := pstoreds.NewPeerstore(ctx, rootstore, pstoreds.DefaultOpts())
-	if err != nil {
-		return nil, err
-	}
-
-	if options.PrivateKey == nil {
-		// generate an ephemeral private key
-		key, err := crypto.GenerateEd25519()
+	peers := make([]peer.AddrInfo, len(options.BootstrapPeers))
+	for i, p := range options.BootstrapPeers {
+		addr, err := peer.AddrInfoFromString(p)
 		if err != nil {
 			return nil, err
 		}
-		options.PrivateKey = key
+		peers[i] = *addr
 	}
```

Comment:
question: This for-loop is never entered in any test, is it possible to test this easily?

Reply:
added a test for the non error case